hail/python/hail/vds/variant_dataset.py

```diff
@@ -7,7 +7,8 @@
 from hail.genetics import ReferenceGenome
 
 
-def read_vds(path, *, intervals=None, n_partitions=None) -> 'VariantDataset':
+def read_vds(path, *, intervals=None, n_partitions=None,
+             _assert_reference_type=None, _assert_variant_type=None) -> 'VariantDataset':
```

Comment:
I don't remember the plan for these -- were you going to implement a checked read in a stacked PR?

Reply:
In many ways, these options are the whole reason for this PR. We want to be able to create a MatrixTable without a blocking round trip to the backend to get it's type. The async PR uses these options.